onnxruntime/core/framework/execution_frame.cc

```diff
@@ -280,7 +280,11 @@ Status ExecutionFrame::AllocateMLValueTensorSelfOwnBufferHelper(OrtValue& ort_va
           return status;
         }
         if (block->size_ != size) {
-          LOGS_DEFAULT(WARNING) << "For ort_value with index: " << ort_value_index
+          // the block size may vary especially if the model has NonZero ops, or different sequence lengths are
```

Comment:
What do you mean "different sequence lengths"? For example, if a model can accept different batch sizes, does it mean memory pattern may not work?

Reply:
It will work in lots of places, just not all. e.g. LSTM and GRU have (optional) outputs that have a dimension of the sequence length, and the size of NonZero output is dependent on the input (e.g. in BiDAF different images will have different sizes for the NonZero node output). So the planning should work up until it hits that sort of node, and fail until it gets back to a point where the output data from a node is not dependent on the sequence length. That's why it would be good to not be as strict on block size matching (a block that is too big can be used).